Source/IGListAutoSectionController.h

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import <IGListKit/IGListMacros.h>
+#import <IGListKit/IGListSectionType.h>
+#import <IGListKit/IGListSectionController.h>
+
+@protocol IGListDiffable;
+@protocol IGListBindable;
+
+/**
+ Other naming options
+ 
+ - IGListDiffingSectionController
+ - IGListBindingSectionController
+ - IGListAutoSectionController
+ */
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol IGListAutoSectionControllerDataSource <NSObject>
+
+- (NSArray<id<IGListDiffable>> *)viewModelsForObject:(id)object;
+- (UICollectionViewCell<IGListBindable> *)cellForViewModel:(id)viewModel atIndex:(NSInteger)index;
+- (CGSize)sizeForViewModel:(id)viewModel;
```

Comment:
Should these pass sender i.e. `- (CGSize)autoSectionController:(IGListAutoSectionController *)sectionController sizeForViewModel:(id)viewModel`? It's more formally correct but uglier.